static/src/js/components/DownloadHistory/DownloadHistory.jsx

```diff
@@ -59,6 +59,15 @@ export const DownloadHistory = ({
     }
   }, [onDeleteRetrieval])
 
+  const handleRemoveClick = useCallback((id) => () => handleRemove(id), [handleRemove])
+
+  // Memoize the descriptions for all items in the history
+  const memoizedDescriptions = useMemo(() => retrievalHistory.reduce((acc, retrieval) => {
```

Comment:
Minor thing but instead of calling this `memoizedDescriptions` which we dont really do for other memoized data, it could just be called `retrievalDescriptions` or something...

```suggestion
  const retrievalDescriptions = useMemo(() => retrievalHistory.reduce((acc, retrieval) => {
```